stdlib/args.ysh

```diff
@@ -60,10 +60,21 @@ proc arg-parse (; place ; ; block_def) {
     setvar p.rest = null
   }
 
+  var names = {}
+  for items in ([p.flags, p.args]) {
+    for x in (items) {
+      if (x.name in names) {
+        error "Duplicate flag/arg name $[x.name] in spec" (status=3)
+      }
+
+      setvar names[x.name] = null
+    }
+  }
+
   call place->setValue(p)
 }
 
-proc flag (short, long ; type='bool', default=null, help=null) {
+proc flag (short, long ; type='bool' ; default=null, help=null, required=false) {
```

Comment:
hm we didn't talk about this but I also think we should allow all of

```
flag --long-only
flag -v
flag -v --verbose
```

I believe this is common, if you look at `grep --help`, etc.  There are some flags that are only long

This requires some more subtle validation, but we can do it with `...word_args`